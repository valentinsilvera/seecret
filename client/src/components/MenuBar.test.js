import React from 'react';
import { render, screen } from '@testing-library/react';
import MenuBar from './MenuBar';
import { BrowserRouter } from 'react-router-dom';

describe('MenuBar', () => {
    const renderComponent = () =>
        render(
            <BrowserRouter>
                <MenuBar />
            </BrowserRouter>
        );

    it("should on initial render, when there's no user, the login and register buttons appear", () => {
        renderComponent();

        expect(screen.getByRole('link', { name: /register/i }));
        expect(screen.getByRole('link', { name: /login/i }));
    });

    it('should match snapshot', () => {
        const { container } = renderComponent();
        expect(container).toMatchSnapshot();
    });
});
