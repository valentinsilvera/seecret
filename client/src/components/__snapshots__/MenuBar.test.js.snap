// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuBar should match snapshot 1`] = `
<div>
  <div
    class="ui blue massive pointing secondary menu"
  >
    <a
      class="active item"
      href="/"
    >
      Seecret ðŸ¤«
    </a>
    <div
      class="right menu"
    >
      <a
        class="item"
        href="/login"
      >
        Login
      </a>
      <a
        class="item"
        href="/register"
      >
        Register
      </a>
    </div>
  </div>
</div>
`;
